operator_ui/src/connectors/redux/reducers/jobRuns.ts

```diff
@@ -1,37 +1,54 @@
 import { RECEIVE_DELETE_SUCCESS } from '../../../actions'
 import pickBy from 'lodash/pickBy'
 
+export interface IState {
+  items: { [k: string]: any }
+  currentPage?: number
+  currentJobRunsCount?: number
+}
+
 const initialState = {
   items: {},
-  currentPage: null,
-  currentJobRunsCount: null
+  currentPage: undefined,
+  currentJobRunsCount: undefined
 }
 
 export const UPSERT_JOB_RUNS = 'UPSERT_JOB_RUNS'
 export const UPSERT_RECENT_JOB_RUNS = 'UPSERT_RECENT_JOB_RUNS'
 export const UPSERT_JOB_RUN = 'UPSERT_JOB_RUN'
 
-export default (state = initialState, action = {}) => {
+export type Action =
+  | { type: 'UPSERT_JOB_RUNS'; data: any }
+  | { type: 'UPSERT_RECENT_JOB_RUNS'; data: any }
+  | { type: 'UPSERT_JOB_RUN'; data: any }
+  | { type: 'RECEIVE_DELETE_SUCCESS'; response: any }
+
+export default (state: IState = initialState, action: Action) => {
   switch (action.type) {
-    case UPSERT_JOB_RUNS: {
+    case 'UPSERT_JOB_RUNS': {
```

Comment:
will using the consts here from lines 16-18 cause type errors with `Action`? those exports are used only in 1 file and other files use string literals, so maybe its better to deprecate the const vars fpor action names entirely?

Reply:
Removed usage of `const`s in the tests